hubblestack/extmods/hubble_mods/curl.py

```diff
@@ -0,0 +1,238 @@
+# -*- encoding: utf-8 -*-
+"""
+Module curl
+=============================
+
+This module allows for querying against URLs
+
+Note that this module doesn't actually shell out to curl. Instead, it uses
+the requests library, primarily for performance concerns.
+
+Also note that this module doesn't support chaining from other modules.
+This is due to security concerns -- because Hubble can collect arbitrary data from
+a system, we don't want an attacker to be able to send that data to arbitrary
+endpoints.
+
+Audit Example:
+---------------
+check_unique_id:
+  description: 'curl check'
+  tag: 'ADOBE-01'
+  implementations:
+    - filter:
+        grains: 'G@osfinger:CentOS*Linux-7'
+      hubble_version: '>3 AND <7 AND <8'
+      module: curl
+      items:
+        - args:
+            url: "https://adobe.com/someurl"
+            function: GET
+            params:
+                key1: val1
+                key2: val2
+            data: "payload for post/put"
+            headers:
+                header1: val1
+                header2: val2
+            username: user
+            password: pwd
+            timeout: 9
+            decode_json: True
+
+          comparator:
+            type: "string"
+            match: "host*"
+            is_regex: true
+
+Note: You can use any comparator depending upon the kind of data a url is returning.
+
+FDG Example:
+------------
+main:
+  description: 'curl check'
+  module: curl
+    args:
+        url: "https://adobe.com/someurl"
+        function: GET
+        params:
+            key1: val1
+            key2: val2
+        data: "payload for post/put"
+        headers:
+            header1: val1
+            header2: val2
+        username: user
+        password: pwd
+        timeout: 9
+        decode_json: True
+
+Mandatory Params:
+    url
+"""
```

Comment:
Please write sample output also

Reply:
it is very much dependent on the URL. Not fixed.